istioctl/cmd/analyze.go

```diff
@@ -339,17 +339,51 @@ func gatherFiles(args []string) ([]local.ReaderSource, error) {
 			}
 			r = os.Stdin
 		} else {
-			r, err = os.Open(f)
+			fm, err := os.Stat(f)
 			if err != nil {
 				return nil, err
 			}
-			runtime.SetFinalizer(r, func(x *os.File) { x.Close() })
+			if fm.IsDir() {
+				dirReaders, err := gatherFilesInDirectory(f)
+				if err != nil {
+					return nil, err
+				}
+				readers = append(readers, dirReaders...)
+			} else {
+				r, err = os.Open(f)
+				if err != nil {
+					return nil, err
+				}
+				runtime.SetFinalizer(r, func(x *os.File) { x.Close() })
+			}
 		}
 		readers = append(readers, local.ReaderSource{Name: f, Reader: r})
 	}
 	return readers, nil
 }
 
+func gatherFilesInDirectory(dir string) ([]local.ReaderSource, error) {
+	var readers []local.ReaderSource
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
```

Comment:
Looks like you need to return `filepath.SkipDir` here to stop filepath.Walk from continuing into subdirectories.

Reply:
Done.